elife_bus_sdk/queues/sqs_queue.py

```diff
@@ -0,0 +1,102 @@
+from typing import Dict, Generator, List
+
+try:
+    import boto3
+except ImportError:  # pragma: no cover
+    # boto3 not yet available, may happen in initial install of elife_bus_sdk package
+    pass
+
+from elife_bus_sdk.queues.message_queue import MessageQueue
+
+
+class SQSMessageQueue(MessageQueue):
+    name = 'sqs'
+
+    def __init__(self, **overrides):
+        self._queue_name = overrides.pop('QueueName', None)
+        self._overrides = overrides
+        self._resource = boto3.resource(self.name, **self._overrides)
+        self._queue = self._resource.get_queue_by_name(QueueName=self._queue_name)
+
+    def dequeue(self) -> List['sqs.Message']:
+        """Retrieves one or more messages (up to 10), from the queue.
+
+        :return: list
+        """
+        conf = self._overrides
+        return self._queue.receive_messages(MaxNumberOfMessages=conf.get('MaxNumberOfMessages', 1),
+                                            VisibilityTimeout=conf.get('VisibilityTimeout', 60),
+                                            WaitTimeSeconds=conf.get('WaitTimeSeconds', 20))
+
+    def enqueue(self, message: str) -> Dict[str, str]:
+        """Delivers a message to the queue.
+
+        :param message: str
+        :return: dict
+
+        example return value:
+        {
+            'MD5OfMessageBody': 'string',
+            'MD5OfMessageAttributes': 'string',
+            'MessageId': 'string',
+            'SequenceNumber': 'string'
+        }
+        """
+        return self._queue.send_message(MessageBody=message)
+
+    @staticmethod
+    def parse_message(message: 'sqs.Message') -> Dict[str, str]:
```

Comment:
never seen an application doing this, found something that accesses fields different from `body`?

Reply:
No, though it seemed a shame to swallow all of the additional information. This is now addressed by returning an `Event` instead. (this is just an internal method now).